source/PyMaterialX/PyMaterialXGenShader/PyShaderPort.cpp

```diff
@@ -18,12 +18,19 @@ void bindPyShaderPort(py::module& mod)
         .def("setName", &mx::ShaderPort::setName)
         .def("getName", &mx::ShaderPort::getName)
         .def("getFullName", &mx::ShaderPort::getFullName)
+        .def("setVariable", &mx::ShaderPort::setVariable)
+        .def("getVariable", &mx::ShaderPort::getVariable)
         .def("setSemantic", &mx::ShaderPort::setSemantic)
         .def("getSemantic", &mx::ShaderPort::getSemantic)
         .def("setValue", &mx::ShaderPort::setValue)
         .def("getValue", &mx::ShaderPort::getValue)
+        .def("setGeomProp", &mx::ShaderPort::setGeomProp)
+        .def("getGeomProp", &mx::ShaderPort::getGeomProp)
         .def("setPath", &mx::ShaderPort::setPath)
         .def("getPath", &mx::ShaderPort::getPath)
         .def("setUnit", &mx::ShaderPort::setUnit)
-        .def("getUnit", &mx::ShaderPort::getUnit);
+        .def("getUnit", &mx::ShaderPort::getUnit)
+        .def("isUniform", &mx::ShaderPort::isUniform)
+        .def("isEmitted", &mx::ShaderPort::isEmitted)
+        .def("isBindInput", &mx::ShaderPort::isBindInput);
```

Comment:
Can you remove isBindInput() this API will be deprecated so shouldn't expose it. Thanks.